state_file.py

```diff
@@ -0,0 +1,67 @@
+import os
+
+class StateFile:
+    default_dict = {
+        "pre-step" :"offline",
+        "masters"  :"offline",
+        "workers"  :"offline",
+        "post-step":"offline",
+        "cluster-name":"",
+    }
+
+    def __init__(self):
+        self.f = open("state_file", "a+")
```

Comment:
Use the cda yaml config to determine where to store it. Provide it with a sane default.